src/kv/value/internal/cast/primitive.rs

```diff
@@ -0,0 +1,250 @@
+// Use consts to match a type with a conversion fn
+// Pros: fast, will work on stable soon (possibly 1.45.0)
+// Cons: requires a `'static` bound
+#[cfg(all(src_build, feature = "kv_unstable_const_primitive"))]
```

Comment:
@hawkw I’ve got a few different strategies for capturing the value from a generic `T` here now.

For `tracing`, if this is something you wanted to do you might end up needing to look more at the specialization one, because the others require a `’static` bound in order to get the type id, which from what I can see isn’t compatible with how the `#[instrument]` macro currently works.